File: Data Structures & Algorithms/zeroOneKnapsack/submission-0.java
class Solution {
    public int maximumProfit(List<Integer> profit, List<Integer> weight, int capacity) {
        return dfs(profit, weight, capacity, 0);
    }

    public int dfs(List<Integer> profit, List<Integer> weight, int capacity, int i) {
        if(i == profit.size()) return 0;

        int maxProfit = dfs(profit, weight, capacity, i+1);

        int newCap = capacity - weight.get(i);
        if(newCap >= 0) {
            int p = profit.get(i) + dfs(profit, weight, newCap, i+1);
            maxProfit = Math.max(p, maxProfit);
        }
        return maxProfit;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = 1
        max_profit = 0
        for r in range(len(prices)):
            profit = prices[r] - prices[l]
            max_profit = max(max_profit, profit)

            if prices[r] < prices[l]:
                l = r
                r += 1
        return max_profit